src/main/java/com/redhat/devtools/lsp4ij/operations/completion/LSPCompletionContributor.java

```diff
@@ -132,9 +132,13 @@ private void addCompletionItems(PsiFile file, Editor editor, CompletionPrefix co
             // Create lookup item
             var lookupItem = createLookupItem(file, editor, completionPrefix.getCompletionOffset(), item, itemDefaults, languageServer);
             // Group it by using completion item kind
-            var groupedLookupItem = PrioritizedLookupElement.withGrouping(lookupItem, item.getKind().getValue());
+            int group = item.getKind() == null ? 0 : item.getKind().getValue();
+            var groupedLookupItem = PrioritizedLookupElement.withGrouping(lookupItem, group);
             // Compute the prefix
-            String prefix = completionPrefix.getPrefixFor(lookupItem.getTextEditRange(), item);
+            String prefix = null;
+            if (lookupItem.getTextEdit() != null) {
```

Comment:
Exposing a new method like getTextEdit just tocheck the null value is shame. IMHO we should continue to use getTextEditRange:

```java
var textEditRange = lookupItem.getTextEditRange();
String prefix = textEditRange != null ? completionPrefix.getPrefixFor(textEditRange, item) : null;
```

getTextEditRange must be updated to return null if item.getTextEdit() is null. If you do that please add Javadoc to this getTextEditRange and anotate `with @Nullable`

```java
public @Nullable Range getTextEditRange() {
    Either<TextEdit, InsertReplaceEdit> textEdit = item.getTextEdit();
    if (textEdit == null) {
        return null;
    }
    if (textEdit.isLeft()) {
        return textEdit.getLeft().getRange();
    } else {
        // here providing insert range, currently do not know if insert or replace is requested
        return textEdit.getRight().getInsert();
    }
}
```

The main idea when we do some PR is to clean the code from LSP4IJ step by step by adding `@NotNull` and `@Nullable` annotations and add Javadoc.

Reply:
Sounds good to me, I was also thinking of something like this.